enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/utils/ArtifactMatcher.java

```diff
@@ -44,9 +44,10 @@ public final class ArtifactMatcher {
      * @author I don't know
      */
     public static class Pattern {
-        private String pattern;
+        private final String pattern;
 
-        private String[] parts;
+        private final String[] parts;
+        private final java.util.regex.Pattern[] partsRegex;
```

Comment:
why not use import here?

Reply:
The inner class this is part of is called `Pattern` so thus was to distinguish between them